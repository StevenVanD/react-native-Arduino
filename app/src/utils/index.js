import { getLabel } from './translations';

export { getLabel };
