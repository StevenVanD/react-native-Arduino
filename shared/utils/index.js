import Colors from './colors';
import Translations from './translations';

export {
  Colors,
  Translations,
};
