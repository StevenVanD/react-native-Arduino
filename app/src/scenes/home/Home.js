import React from "react";
import { ActivityIndicator, View, FlatList } from "react-native";
import PropTypes from "prop-types";
import styles from "./styles";
import { Colors } from "../../../../shared/utils";
import { Translations } from "../../utils";
import { Text, ListItem } from "../../components";

class Home extends React.Component {
  renderListItem = () => {
    return (<ListItem />);
  }
  render() {
    return (
      <View style={styles.container}>
        <Text center style={styles.lead}>
          hey
        </Text>
        <FlatList
          data={["dfgdfg", "zerzer", "sdffds"]}
          renderItem={this.renderListItem}
        />
      </View>
    );
  }
}

Home.propTypes = {};

Home.defaultProps = {};

export default Home;