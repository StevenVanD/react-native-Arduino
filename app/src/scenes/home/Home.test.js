import React from 'react';
import { shallowTestHelper, eventChangeHelper } from '../../utils/test';
import Home from './Home';

const mockFn = jest.fn();
const component = <Home />;
//eventChangeHelper({ component, mockFn });
shallowTestHelper(component);