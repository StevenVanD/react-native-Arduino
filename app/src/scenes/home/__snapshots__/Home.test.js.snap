// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow test should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Text
          center={true}
          label=""
          style={
            Object {
              "paddingVertical": 20,
            }
          }
          uppercase={false}
        >
          hey
        </Text>,
        <FlatList
          data={
            Array [
              "dfgdfg",
              "zerzer",
              "sdffds",
            ]
          }
          disableVirtualization={false}
          horizontal={false}
          initialNumToRender={10}
          keyExtractor={[Function]}
          maxToRenderPerBatch={10}
          numColumns={1}
          onEndReachedThreshold={2}
          renderItem={[Function]}
          scrollEventThrottle={50}
          updateCellsBatchingPeriod={50}
          windowSize={21}
        />,
      ],
      "style": Object {
        "display": "flex",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "space-between",
        "padding": 32,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "center": true,
          "children": "hey",
          "label": "",
          "style": Object {
            "paddingVertical": 20,
          },
          "uppercase": false,
        },
        "ref": null,
        "rendered": "hey",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "data": Array [
            "dfgdfg",
            "zerzer",
            "sdffds",
          ],
          "disableVirtualization": false,
          "horizontal": false,
          "initialNumToRender": 10,
          "keyExtractor": [Function],
          "maxToRenderPerBatch": 10,
          "numColumns": 1,
          "onEndReachedThreshold": 2,
          "renderItem": [Function],
          "scrollEventThrottle": 50,
          "updateCellsBatchingPeriod": 50,
          "windowSize": 21,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Text
            center={true}
            label=""
            style={
              Object {
                "paddingVertical": 20,
              }
            }
            uppercase={false}
          >
            hey
          </Text>,
          <FlatList
            data={
              Array [
                "dfgdfg",
                "zerzer",
                "sdffds",
              ]
            }
            disableVirtualization={false}
            horizontal={false}
            initialNumToRender={10}
            keyExtractor={[Function]}
            maxToRenderPerBatch={10}
            numColumns={1}
            onEndReachedThreshold={2}
            renderItem={[Function]}
            scrollEventThrottle={50}
            updateCellsBatchingPeriod={50}
            windowSize={21}
          />,
        ],
        "style": Object {
          "display": "flex",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "space-between",
          "padding": 32,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "center": true,
            "children": "hey",
            "label": "",
            "style": Object {
              "paddingVertical": 20,
            },
            "uppercase": false,
          },
          "ref": null,
          "rendered": "hey",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "data": Array [
              "dfgdfg",
              "zerzer",
              "sdffds",
            ],
            "disableVirtualization": false,
            "horizontal": false,
            "initialNumToRender": 10,
            "keyExtractor": [Function],
            "maxToRenderPerBatch": 10,
            "numColumns": 1,
            "onEndReachedThreshold": 2,
            "renderItem": [Function],
            "scrollEventThrottle": 50,
            "updateCellsBatchingPeriod": 50,
            "windowSize": 21,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
